go/tools/releaser/upgradedep_test.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	bzl "github.com/bazelbuild/buildtools/build"
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	reposFile = "go/tools/releaser/testdata/repositories.bzl"
+	funcName  = "go_rules_dependencies"
+)
+
+func TestPatchItemParser(t *testing.T) {
```

Comment:
This test adds a lot of complexity just to cover whether whether a patch expression is parsed correctly. Can we avoid all the file loading and rule parsing in the test by doing something like:

```go
patchExpr, err := bzl.Parse("repos.bzl", 
[]byte(`# releaser:patch-cmd gazelle -repo_root . -go_prefix golang.org/x/tools -go_naming_convention import_alias
 Label("//third_party:org_golang_x_tools-gazelle.patch")`))
if err != nil {
   t.Fatalf(...)
}
patchLabelStr, comment, err := parsePatchesItem(patchExpr)
if patchLabelStr != "//third_party:org_golang_x_tools-gazelle.patch" {
  t.Fatalf(...)
}
```

Reply:
Oh that makes sense. Didn't immediately think that we don't really need all the wrapping around the test. I'll put something together in the AM.